src/test_lib_json/main.cpp

```diff
@@ -3903,6 +3903,37 @@ JSONTEST_FIXTURE_LOCAL(FuzzTest, fuzzDoesntCrash) {
                              example.size()));
 }
 
+struct GetLocationLineAndColumnTest : JsonTest::TestCase {
+  void testLineAndColumn(const std::string& doc, ptrdiff_t location, int row,
+                         int column) {
+    Json::LineAndColumn actualLocation = Json::getLocationLineAndColumn(
+        doc.data(), doc.data() + doc.length(), location);
+    JSONTEST_ASSERT_EQUAL(row, actualLocation.line);
+    JSONTEST_ASSERT_EQUAL(column, actualLocation.column);
+  }
+};
+
+JSONTEST_FIXTURE_LOCAL(GetLocationLineAndColumnTest, test) {
+  const std::string example = "line 1\nline 2\r\nline 3\rline 4";
+  size_t found_l = -1;
+  size_t found_i = -1;
+  size_t found_n = -1;
+  size_t found_e = -1;
+  size_t found_space = -1;
```

Comment:
This test is too clever I think. Trying to reuse these variables and use a fixture function.
It can just hard-code these positions and use a more creative test string.
Just hard-code some inputs and outputs in a table, and run a loop over that table.
It would be less code and easier to maintain.

Improve the name "test".

Needs more coverage.
I think it's missing scenarios where:
    - input has trailing CR,LF,CRLF at end of string or beginning.
    - input with blank lines.
    - input with embedded `\0`.
    - weird stuff like LFCR or CRCRLF.